gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -181,7 +157,38 @@ case class OffloadJoin() extends OffloadSingleNode with LogLevelUtil {
       case other => other
     }
   }
+}
+
+case class OffloadSortMergeJoin() extends OffloadSingleNode with LogLevelUtil {
+  override def offload(plan: SparkPlan): SparkPlan = {
+    if (TransformHints.isNotTransformable(plan)) {
+      logDebug(s"Columnar Processing for ${plan.getClass} is under row guard.")
+      return plan
+    }
+    plan match {
+      case plan: SortMergeJoinExec =>
+        OffloadSortMergeJoin.transformSortMergeJoinExec(plan)
```

Comment:
I mean we can add a rule to convert `SortMergeJoinExec` to `ShuffledHashJoinExec`. So that we do not need to check  `forceShuffledHashJoin` here and there.